### PR TITLE
CI: Switch from travis to Github actions for tests

### PR DESCRIPTION
**Pull request by rossbar**
Moves testing that was previously done on TravisCI to github actions. A few things to note:
 - I erred on the side of over-testing in specifying the testing platforms: Python versions [3.6 through 3.9] cross sphinx versions [1.6.5, 2.1, latest]. These sphinx versions were tested explicitly in the Travis job. This testing matrix is probably overkill, so please feel free to remove Python and/or Sphinx versions as you see fit.
 - TravisCI had previously built the documentation (presumably to test for doc-build errors w/ different sphinx versions). I add this to the GH action as well, though perhaps it's unnecessary and the circleCI doc build job is sufficient.

### CODE REVIEW

**Comment by rossbar** [comment]
Also I'm not quite sure how to setup codecov properly - help there would be appreciated!

**Comment by larsoner** [comment]
[I think](https://github.com/mne-tools/mne-python/blob/master/.github/workflows/linux_pip.yml#L69-L71) it's just:
```
      - uses: codecov/codecov-action@v1
        if: success()
        name: 'Upload coverage to CodeCov'
```

**Comment by rossbar** [comment]
Okay, I added a new action to run test coverage so that it wasn't run for every single one of the jobs in the testing matrix. With some minor tweaks to the pytest-cov CLI options and `.coveragerc` I was able to get the coverage running and uploading again. I'm not sure where the drop in coverage relative to master is coming from, maybe I'm calling pytest with the pytest-cov plugin incorrectly?

**Comment by larsoner** [comment]
> I'm not sure where the drop in coverage relative to master is coming from, maybe I'm calling pytest with the pytest-cov plugin incorrectly?

Probably because you made one separate `coverage` run rather than having all runs (different Python versions and especially different Sphinx versions) report coverage. It's probably better just to let them all report coverage and let `codecov` combine them.

File: .travis.yml [hunk 1]
```diff
@@ -1,54 +0,0 @@
...
-  # Build documentation
-  - |
-    cd ../doc
-    make SPHINXOPTS=$SPHINXOPTS html
-    make SPHINXOPTS=$SPHINXOPTS latexpdf
-
-after_script:
-  - |
-    cd ../dist
-    codecov
```
Comment (larsoner) [inline .travis.yml:54]: ... previously all runs reported coverage

**Comment by rossbar** [comment]
Thanks for the suggestion on coverage for the whole test matrix - I didn't realize codecov handled that so cleanly.

Unfortunately it didn't fix the drop in coverage issue. I've taken a closer look at the coverage reports themselves, and it seems the drop is entirely related to the fact that the coverage is *now* reporting partial lines when it wasn't before. For example, compare the coverage report from [an older PR](https://codecov.io/gh/numpy/numpydoc/compare/05e5c2b2efd24de9a6942d07cb90f341217c22ae...651edbd1eb75a3ef3ed99d1504a8b9927d07d0c4/src/numpydoc/docscrape.py) to that of [this PR](https://codecov.io/gh/numpy/numpydoc/compare/bb96d0fca570cec533251ef4c0f309020a5dca04...e005cb63294675178a81e93e88af1bdc47f09f36/src/numpydoc/docscrape.py). The [coverage summary](https://codecov.io/gh/numpy/numpydoc/compare/bb96d0fca570cec533251ef4c0f309020a5dca04...e005cb63294675178a81e93e88af1bdc47f09f36/changes) for this PR also seems to indicate that the only changes are related to partially covered branches, rather than any new missing coverage.

AFAICT there is no coverage option to enable/disable the partial coverage reporting, so I'm not sure why the partial coverage is included here and not in the older reports. IMO if the only difference is stricter reporting of partially-covered branches, that might actually be considered an improvement. I'd have to check all the coverage reports for each file to confirm that this is indeed the case though.

**Comment by larsoner** [comment]
Thanks @rossbar. Not sure why coverage is wacky but I think this is good enough to merge. We can always improve coverage itself and coverage reporting in additional PRs if we need to.

**Pull request merged**